tests/lib/rule-tester/rule-tester.js

```diff
@@ -2295,6 +2228,200 @@ describe("RuleTester", () => {
         });
     });
 
+    describe("deprecations", () => {
+        let processStub;
+        const ruleWithNoSchema = {
```

Comment:
There needs to be tests with `schema: null` and `schema: undefined` as well.

Reply:
Added in https://github.com/eslint/eslint/pull/16063/commits/b23b8c13c2ad731b2b30d810b2928d2c84c2b035